rfc/rfc018-openid-credential-issuance.md

```diff
@@ -0,0 +1,140 @@
+
+## 3.1 Supported Flows and Features
+
+Conforming implementations MUST support the following features:
+
+- Pre-authorized code flow
+- Credential offer:
+  - with `credential_offer` parameter
+  - with `string` values, referring to a credential's `id` in `credentials_supported`
+- Credentials in JSON-LD format
+- Issuer- and wallet metadata
+
+## 3.2 Credential Offer Response
+
+The OpenID4VCI specification does not specify responses for the credential offer.
+But since this specification describes server-to-server interaction
+a response is useful for the issuer to know whether the credential offer was accepted.
+
+### 3.2.1 Credential Offer Success Response
+
+In case the wallet considers the offer valid, it MUST respond with HTTP status `200 OK`.
+This is regardless of whether the wallet will actually revrieve the wallet.
+
+If the wallet decides to retrieve the credential synchronously, it SHOULD respond with `status` set to `credential_received`.
+Below is a non-normative example of such a response:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "status": "credential_received"
+}
+```
+
+### 3.2.2 Credential Offer Error Response
+
+This specification extends the specification with the following error responses a wallet MUST support when receiving a credential offer:
+
+- `invalid_request`: the offer is missing or invalid.
+- `invalid_grant`: the offer does not contain a grant the wallet could use to retrieve the credential.
+
+It MUST respond with HTTP status `400 Bad Request` and a response body containing the error code.
+
+Below is a non-normative example of such a response:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "error": "invalid_grant"
+}
+```
+
+## 4. Client Metadata Discovery using DID documents
+
+To offer a credential the issuer needs to discover the wallet's credential offer endpoint, which is registered in the client metadata.
+
+To allow a (Nuts) wallet to be discovered, the wallet MUST be represented by a resolvable DID.
+The DID document MUST contain a `node-http-services-baseurl` service, which contains the base URL for node-to-node HTTP services.
+The issuer appends the following path to it to resolve the client metadata: `/n2n/identity/<did>/.well-known/openid-credential-wallet`,
+replacing `<did>` with the DID of the wallet holder.
+
+## 5. Security
+
+### 5.1 Credential Issuer Trust
+
+In server-to-server issuance of credentials, there's no user approving the credential offer before requesting it and loading it into the holder's wallet.
+Therefore, the holder must make a decision whether to trust the credential issuer.
+Trust SHOULD be derived from the TLS trust anchors referred to by section 5.2 (Transport Security).
+
+### 5.2 Transport Security
+
+To prevent against eavesdropping and man-in-the-middle attacks, all exchanges MUST be done over HTTPS.
```

Comment:
This is already a hard requirement derived from the OAuth/OpenID specs that the OpenID4VCI spec builds on. Maybe we should mention this since this spec only defines what we added?